cmd/fpga_admissionwebhook/patcher.go

```diff
@@ -169,58 +182,94 @@ func (p *patcher) translateFpgaResourceName(oldname corev1.ResourceName) (string
 	return "", errors.Errorf("Unknown FPGA resource: %s", rname)
 }
 
+func (p *patcher) checkResourceRequests(container corev1.Container) error {
+	for resourceName, resourceQuantity := range container.Resources.Requests {
+		interfaceID, _, err := p.parseResourceName(string(resourceName))
+		if err != nil {
+			return err
+		}
+
+		if interfaceID == "" {
+			// Skip non-FPGA resources
+			continue
+		}
+		if container.Resources.Limits[resourceName] != resourceQuantity {
+			return errors.Errorf("'limits' and 'requests' for %s must be equal", string(resourceName))
+		}
+	}
+
+	return nil
+}
+
 func (p *patcher) getPatchOpsOrchestrated(containerIdx int, container corev1.Container) ([]string, error) {
 	var ops []string
 
-	mutated := false
+	for _, v := range container.Env {
+		if strings.HasPrefix(v.Name, "FPGA_REGION") || strings.HasPrefix(v.Name, "FPGA_AFU") {
+			return nil, errors.Errorf("The environment variable '%s' is not allowed", v.Name)
+		}
+	}
+
+	if err := p.checkResourceRequests(container); err != nil {
+		return nil, err
+	}
+
+	regions := make(map[string]int64)
+	envVars := make(map[string]string)
+	counter := 0
 	for resourceName, resourceQuantity := range container.Resources.Limits {
 		interfaceID, afuID, err := p.parseResourceName(string(resourceName))
 		if err != nil {
 			return nil, err
 		}
 
 		if interfaceID == "" && afuID == "" {
+			// Skip non-FPGA resources
 			continue
 		}
 
-		if mutated {
-			return nil, errors.Errorf("Only one FPGA resource per container is supported in '%s' mode", orchestrated)
+		if container.Resources.Requests[resourceName] != resourceQuantity {
+			return nil, errors.Errorf("'limits' and 'requests' for %s must be equal", string(resourceName))
 		}
 
-		op := fmt.Sprintf(resourceReplaceOp, containerIdx, "limits", rfc6901Escaper.Replace(string(resourceName)),
-			containerIdx, "limits", rfc6901Escaper.Replace(namespace+"/region-"+interfaceID), resourceQuantity.String())
-		ops = append(ops, op)
+		quantity, ok := resourceQuantity.AsInt64()
+		if !ok {
+			return nil, errors.New("Resource quantity isn't of integral type")
+		}
+		regions[interfaceID] = regions[interfaceID] + quantity
 
-		if afuID != "" {
-			oldVars, err := getEnvVars(container)
-			if err != nil {
-				return nil, err
-			}
-			op = fmt.Sprintf(envAddOp, containerIdx, interfaceID, afuID, oldVars)
-			ops = append(ops, op)
+		for i := int64(0); i < quantity; i++ {
```

Comment:
why int64 is used here?

Reply:
Because `quantity` is of `int64`. By default Go infers `i` as `int` in the expression `i := 0` and my Go compiler refused to compare values of different types:
```
go build -v && go test -v && go vet
github.com/intel/intel-device-plugins-for-kubernetes/cmd/fpga_admissionwebhook
# github.com/intel/intel-device-plugins-for-kubernetes/cmd/fpga_admissionwebhook
./patcher.go:241:17: invalid operation: i < quantity (mismatched types int and int64)
```